File: data_structures_by_c++/arrays/array_deletion.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    // ios::sync_with_stdio(false);
    // cin.tie(nullptr);
    
    int n;
    cout << "Enter number of elements: ";
    cin >> n;
    int arr[100];
    cout << "Enter elements: " << endl;
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    int pos;
    cout << "Enter position to delete (0-based index): ";
    cin >> pos;
    for (int i = pos; i < n - 1; i++)
    {
        arr[i] = arr[i + 1];
    }
    n--;
    cout << "Array after deletion: ";
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
}

File: data_structures_by_python/arrays/array_deletion.py
# ## with stl
# n = int(input("Enter number of elements: "))
# arr = []
# print("Enter elements: ")
# for i in range(n):
#     arr.append(int(input()))

# pos = int(input("Enter position to delete(0 based index): "))
# arr.pop(pos)
# print("After deleting the element: ", arr)


## without stl
n = int(input("Enter number of elements: "))
arr = []
# print("Enter elements: ")
for i in range(n):
    arr.append(int(input()))
pos = int(input("Enter position to delete(0 based index): "))
i = pos
while i < n - 1:
    arr[i] = arr[i + 1]
    i += 1
    
n -= 1
print("After deleting last element:", end=" ")
for i in range(n):
    print(arr[i], end=" ")